Update package manager instructions with community-hosted repos

xmudrii opened the pull request:
This PR documents usage of the community-hosted repositories, implemented as part of [KEP-1731](https://github.com/kubernetes/enhancements/issues/1731).

All documents that mention the deprecated Google-hosted repository are updated to:

- Highlight that the Google-hosted repository is deprecated and that using the community-hosted repositories is strongly recommended
- Highlight the most important differences between the Google-hosted repository and the community-hosted repositories
- Show how to setup both the community-hosted repositories and the Google-hosted repository

I decided to go with `Google-hosted repository` and `Community-hosted repositories` for wording to distinct between repositories hosted on Google and our infra.

It's important to note that the infrastructure and repositories, and the KEP, don't have the same graduation criteria. We consider the infrastructure and repositories to be GA, but the KEP itself is graduating to alpha. That's because KEP covers more than the infrastructure and repositories and those tasks are still to be done in the upcoming release cycles.

xmudrii commented:
@kubernetes/sig-release-leads @kubernetes/release-engineering @sftim I'm looking for some initial feedback on structure and wording. I'll work on the document explaining community-hosted repositories tomorrow.

xmudrii commented:
cc @kubernetes/sig-release-leads @kubernetes/release-engineering @sftim

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
 
+{{< warning >}}
+Kubernetes has two different package repositories starting from August 2023.
+The Google-hosted repository is considered deprecated and we will stop
+publishing packages to that repository starting with Kubernetes 1.30.0.
```
Is this an absolute commitment, or just likely? Only say when we'll stop if we're 100% committed.

xmudrii replied:
@sftim This is something to verify with leads but most of them are not available this week. I'll make sure to follow up on this early next week.

xmudrii replied:
I removed the timeline but still left a note that this will happen in the future.

xmudrii replied:
We decided that Kubernetes 1.29 is going to be cutover. https://kubernetes.slack.com/archives/CCK68P2Q2/p1691085627058939

sftim reviewed (commented):
Hi. Some thoughts; hope these are useful.

Please label this as [WIP] if it is not ready yet to merge.

xmudrii commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+You can create this directory if you need to, making it world-readable but writeable only by admins.
+{{< /note >}}
+
+### Google-hosted repository (Deprecated)
```
> Should we change the name of the file within /etc/apt/keyrings? Eg to /etc/apt/keyrings/kubernetes-deprecated-apt-keyring.gpg

I'd avoid doing this so that we have universal file location for the old key for all users in case we ever need to get back to it, e.g. to remove it if needed. I used a new location for the new key, I think that's kind of a better approach.

xmudrii commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
````diff
@@ -138,20 +149,70 @@ The following methods exist for installing kubectl on Linux:
    ```
 
-   If you use Debian 9 (stretch) or earlier you would also need to install `apt-transport-https`:
+3. Add the appropriate community-hosted Kubernetes `apt` repository. If you want to use Kubernetes version different than {{< param "version" >}},
````
I'm wondering if this is worth it. It is really about only changing one number in the given command which should be intuitive enough given that an example for the current version is provided. Looking from the user/reader perspective, switching to docs version, having to navigate again to the correct section, and reading it all over again is more complicated IMO.

xmudrii commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
```diff
@@ -138,20 +149,70 @@ The following methods exist for installing kubectl on Linux:
 {{< tabs name="kubectl_install" >}}
 {{% tab name="Debian-based distributions" %}}
 
+### Community-hosted repositories
```
@sftim I'm not really sure I understand this comment, can you please elaborate?

xmudrii commented on content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md:
````diff
@@ -52,6 +52,148 @@ The upgrade workflow at high level is the following:
+nano /etc/yum.repos.d/kubernetes.repo
+```
+
+You will see two lines with URLs that contain your current Kubernetes
+minor version. For example, if you're using v{{< skew currentVersionAddMinor -1 "." >}},
+you should see this:
````
@sftim A shortcode would be definitely helpful. Is this something that you can help with?

xmudrii commented on api-ref-generator:
@sftim Any idea why this change suddenly appeared and how to get rid of it?

sftim replied:
This is probably from a rebase or something.

I bet it's hard to explain a least-intervention fix. Try a version of the https://xkcd.com/1597/ fix:
- **copy all the files from this PR somewhere safe**
- reset your `obs` branch to match _dev-1.28_, eg `git checkout obs && git reset --hard upstream/dev-1.28`
   - you can even delete the `obs` branch locally and then make a new one
- copy the files back from where you kept them safe
- commit
- <del>non</del>profit

xmudrii commented on content/en/docs/tasks/administer-cluster/kubeadm/upgrading-linux-nodes.md:
```diff
@@ -17,6 +17,148 @@ upgrade the control plane nodes before upgrading your Linux Worker nodes.
 
 <!-- steps -->
 
+## Changing the package repository
```
@sftim Great point, added a dedicated task page

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+
+<!-- overview -->
+
+This page explains how to switch from one to another Kubernetes package repository
+upon upgrading from one to another Kubernetes minor release. Unlike Google-hosted
```
```suggestion
This page explains how to switch from one package repository to another
when upgrading Kubernetes minor releases. Unlike Google-hosted
```

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+see the ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](TBD)
+blog post.
+
+## {{% heading "prerequisites" %}}
```
I think you can simplify the text by merging the _prerequisites_ and the first step.

Think about it this way: if the first step is to check if the repos are in use, then it is not really a prerequisite to start the procedure.

sftim replied:
We should keep this heading. Maybe the first step becomes a subheading?

xmudrii replied:
> Maybe the first step becomes a subheading?

I like this idea!

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+## Switching to another Kubernetes package repository
+
+This step should be done upon upgrading from one to another Kubernetes minor
+release in order to get access to packages for the desired Kubernetes minor
```
```suggestion
release in order to get access to the packages of the desired Kubernetes minor
```

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+{{< tabs name="k8s_install_versions" >}}
+{{% tab name="Ubuntu, Debian or HypriotOS" %}}
+
+Print contents of the file that defines the Kubernetes `apt` repository:
```
```suggestion
Print the contents of the file that defines the Kubernetes `apt` repository:
```

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+{{% /tab %}}
+{{% tab name="CentOS, RHEL or Fedora" %}}
+
+Print contents of the file that defines the Kubernetes `yum` repository:
```
```suggestion
Print the contents of the file that defines the Kubernetes `yum` repository:
```

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
````diff
@@ -0,0 +1,168 @@
+exclude=kubelet kubeadm kubectl cri-tools kubernetes-cni
+```
+
+3. Save the file and exit your text editor. Proceed with following the relevant upgrade guidelines.
````
```suggestion
3. Save the file and exit your text editor. Continue following the relevant upgrade guidelines.
```

puerco commented on content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade.md:
```diff
@@ -52,6 +52,13 @@ The upgrade workflow at high level is the following:
 
+## Changing the package repository
+
+If you're using the Kubernetes community-owned repositories, you need to change
```
nit: at some point, you can economize on the number of words by switching to just "community repositories". (But maybe I'm biased after reading all docs at once)  
```suggestion
If you're using the community repositories, you need to change
```

sftim replied:
We like to be verbose. For example, it helps readers who don't speak English natively, and who don't know Kubernetes very well.

Even so, I'm -0 on this one (I don't object!)

xmudrii replied:
I agree with @sftim here. Additionally, it might make it easier for users who might not have enough context on how our repositories work.

puerco commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
````diff
@@ -181,15 +180,28 @@ You can create this directory if you need to, making it world-readable but write
+```
+
+{{< note >}}
+If you desire to upgrade kubectl to another minor release at some point, you'll need to change version in `/etc/yum.repos.d/kubernetes.repo` before running `yum update`. This procedure is described in [Changing the Kubernetes package repository](/docs/tasks/administer-cluster/kubeadm/change-package-repository/) document.
````
```suggestion
To upgrade kubectl to another minor release, you'll need to bump the version in `/etc/yum.repos.d/kubernetes.repo` before running `yum update`. This procedure is described in more detail in [its own document](/docs/tasks/administer-cluster/kubeadm/change-package-repository/).
```

puerco reviewed (commented):
Nice work Marko, thank you! I left some suggestions below 👇

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+- The Kubernetes package repositories contain only packages for Kubernetes
+  versions starting with v1.24.0. Earlier versions are available only in the
+  Google-hosted repository.
+- There's a dedicated package repository for each Kubernetes minor version.
+  Please take this into the consideration when upgrading to the next minor
+  release.
```
The original text is more idiomatic English. I like two separate sentences. Although, in tech docs we avoid “Please”.
```suggestion
There's a dedicated package repository for each Kubernetes minor version.
When upgrading to to a different minor release, you must bear in mind that
the package repository details also change.
```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+  Please take this into the consideration when upgrading to the next minor
+  release.
+
+We recommend checking out our official announcement ["pkgs.k8s.io: Introducing
+Kubernetes community-owned package repositories"](TBD) for more information
+about the new Kubernetes package repositories.
```
Try something like:
```diff
-We recommend checking out our official announcement ["pkgs.k8s.io: Introducing
-Kubernetes community-owned package repositories"](TBD) for more information
-about the new Kubernetes package repositories.
+To learn more, you can read the official announcement ["pkgs.k8s.io: Introducing
+Kubernetes community-owned package repositories"](TBD).
```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
 {{% tab name="Debian-based distributions" %}}
 
+### Kubernetes package repositories {#dpkg-k8s-package-repo}
+
```
```suggestion

These instructions are for Kubernetes {{< skew currentVersion >}}.

```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+{{< /note >}}
+
+### Google-hosted package repository (deprecated) {#dpkg-google-package-repo}
+
```
```suggestion

These instructions are for Kubernetes {{< skew currentVersion >}}.

```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+   ```
+
+{{< note >}}
+In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default.
+You can create this directory if you need to, making it world-readable but writeable only by admins.
````
```suggestion
In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default;
you can create it by running `sudo mkdir -m 755 /etc/apt/keyrings`
```
(to match recent changes to _main_)

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
    ```
+
 {{< note >}}
 In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default.
 You can create this directory if you need to, making it world-readable but writeable only by admins.
````
```suggestion
In releases older than Debian 12 and Ubuntu 22.04, `/etc/apt/keyrings` does not exist by default;
you can create it by running `sudo mkdir -m 755 /etc/apt/keyrings`
```
(to match recent changes to _main_)

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+
+{{< note >}}
+- Setting SELinux in permissive mode by running `setenforce 0` and `sed ...` effectively disables it.
+This is required to allow containers to access the host filesystem, which is needed by pod networks for example.
+You have to do this until SELinux support is improved in the kubelet.
```
```suggestion
  This is required to allow containers to access the host filesystem; for example, some cluster
  network plugins require that.
  You have to do this until SELinux support is improved in the kubelet.
```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+sudo sed -i 's/^SELINUX=enforcing$/SELINUX=permissive/' /etc/selinux/config
+```
+
+{{< note >}}
````
This is relevant to security. Use a `caution` callout.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+
+### Prerequisites
+
+1. Set SELinux in the permissive mode:
```
```suggestion
1. Set SELinux to `permissive` mode:
```

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
 {{% tab name="Red Hat-based distributions" %}}
-```bash
+
+### Prerequisites
````
```suggestion
### {{% heading "prerequisites" %}} {#prerequisites-rpm}
```
(untested)


Another good option: go up to line 19 and add some details there, maybe with a link to a heading later in the file.

xmudrii replied:
I dropped this paragraph, I think it should be good enough without it, but let me know if you have any objections to this.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+```
+
+{{< note >}}
+If the `baseurl` fails because your Red Hat-based distribution cannot interpret `basearch`, replace `\$basearch` with your computer's architecture.
````
Tweaked:
```suggestion
If the `baseurl` fails because your RPM-based distribution cannot interpret `$basearch`, replace `\$basearch` with your computer's architecture.
```

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+This document assumes that you're already using the Kubernetes community-owned
+package repositories. If that's not the case, it's strongly recommend to migrate
+from the Google-hosted repository to the Kubernetes package repositories
+as described in [the blog post](TBD).
```
FIXME!

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
````diff
@@ -0,0 +1,168 @@
+cat /etc/apt/sources.list.d/kubernetes.list
+```
+
+If you see the URL similar to:
````
```suggestion
If you see a line similar to:
```

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+
+**You're using the Kubernetes package repositories and this guide applies to you.**
+Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
+as described in [the blog post](TBD).
```
FIXME

xmudrii replied:
Well aware of it, I still didn't receive date for blog post which is part of the URL, but as soon as I do, I'll update this. :)

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
````diff
@@ -0,0 +1,168 @@
+Print contents of the file that defines the Kubernetes `yum` repository:
+
+```shell
+cat /etc/yum.repos.d/kubernetes.repo
````
```suggestion
# On your system, this configuration file could have a different name
cat /etc/yum.repos.d/kubernetes.repo
````

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
````diff
@@ -0,0 +1,168 @@
+Print contents of the file that defines the Kubernetes `apt` repository:
+
+```shell
+cat /etc/apt/sources.list.d/kubernetes.list
````
(nit / preference)
```suggestion
pager /etc/apt/sources.list.d/kubernetes.list
```

Debian provides `pager` as an abstraction over the user's preferred pager.

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,168 @@
+---
+title: Changing the Kubernetes package repository
+min-kubernetes-server-version: 1.24
```
Why this line?

sftim commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
```diff
@@ -22,9 +22,9 @@ Using the latest compatible version of kubectl helps avoid unforeseen issues.
-- [Install kubectl binary with curl on Linux](#install-kubectl-binary-with-curl-on-linux)
-- [Install using native package management](#install-using-native-package-management)
-- [Install using other package management](#install-using-other-package-management)
+ - [Install kubectl binary with curl on Linux](#install-kubectl-binary-with-curl-on-linux)
+ - [Install using native package management](#install-using-native-package-management)
+ - [Install using other package management](#install-using-other-package-management)
```
Why the leading whitespace? I'd leave it out.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+enabled=1
+gpgcheck=1
+gpgkey=https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+exclude=kubelet kubeadm kubectl
```
Maybe comment this line to explain why we configure excludes.

xmudrii replied:
Added it above the code block.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
+[kubernetes]
+name=Kubernetes
+baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-\$basearch
```
I prefer to specify `\$basearch` - it's a nice clue for people scripting things.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
 
-  - You can leave SELinux enabled if you know how to configure it but it may require settings that are not supported by kubeadm.
+```shell
+cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
````
Optionally:
```suggestion
# This overwrites any existing configuration in /etc/yum.repos.d/kubernetes.repo
cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
```

xmudrii replied:
Added to other relevant documents as well.

sftim commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
````diff
@@ -173,44 +230,83 @@ For more information on version skews, see:
+1. Add the Kubernetes `yum` repository:
+
+```shell
 cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
````
Optionally:
```suggestion
# This overwrites any existing configuration in /etc/yum.repos.d/kubernetes.repo
cat <<EOF | sudo tee /etc/yum.repos.d/kubernetes.repo
```

sftim reviewed (commented):
More feedback. Feel very free to skip things marked as nits, optional, or my personal preference.

xmudrii commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,9 +144,66 @@ For more information on version skews, see:
+
+There are some important considerations for the Kubernetes package repositories:
+
+- The Kubernetes package repositories contain only packages for Kubernetes
+  versions starting with v1.24.0. Earlier versions are available only in the
```
> How feasible is it to add in the older packages? That'd be nice context to know (is it impossible, or just effort we haven't staffed?)

It's a manual process that takes a lot of time. I think it's safe to estimate around 45 minutes to 1 hour for one patch release in the best case. I think that given the current state of tooling and not being able to automate this properly, only doing this for supported releases at this time is the best we can do.

xmudrii commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
````diff
@@ -145,28 +145,27 @@ The following methods exist for installing kubectl on Linux:
    ```shell
    sudo apt-get update
-   sudo apt-get install -y ca-certificates curl
+   sudo apt-get install -y apt-transport-https ca-certificates curl
````
Added to other relevant documents as well.

xmudrii commented on content/en/docs/tasks/tools/install-kubectl-linux.md:
````diff
@@ -145,28 +145,27 @@ The following methods exist for installing kubectl on Linux:
-   echo "deb [signed-by=/etc/apt/keyrings/kubernetes-archive-keyring.gpg] https://apt.kubernetes.io/ kubernetes-xenial main" | sudo tee /etc/apt/sources.list.d/kubernetes.list
-   ```
+   {{< note >}}
+   If you desire to upgrade kubectl to another minor release at some point, you'll need to change version in `/etc/apt/sources.list.d/kubernetes.list` before running `apt-get update` and `apt-get upgrade`. This procedure is described in [Changing the Kubernetes package repository](/docs/tasks/administer-cluster/kubeadm/change-package-repository/) document.
````
I applied the suggestion that @puerco proposed.

xmudrii commented:
@sftim @puerco I addressed all your comments and I fixed some observed issues:

- I fixed upgrade commands to accommodate different revision formats between the old repo and the new repos
- I fixed command for adding the new apt repo to include `[signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg]`

All instructions are tested to make sure that everything works as expected.

The TBD links are replaced with the real links, given that
/hold cancel

sftim commented:
/sig release

jeremyrickard commented:
/lgtm

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+there's a dedicated package repository for each Kubernetes minor version.
+
+For more information about the Kubernetes community-owned package repositories,
+see ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).
```
We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

xmudrii replied:
Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+This document assumes that you're already using the Kubernetes community-owned
+package repositories. If that's not the case, it's strongly recommend to migrate
+to the Kubernetes package repositories as described in
+["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).
```
We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

xmudrii replied:
Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+
+**You're using the Kubernetes package repositories and this guide applies to you.**
+Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
+as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).
```
We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

xmudrii replied:
Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

sftim commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+
+**You're using the Kubernetes package repositories and this guide applies to you.**
+Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
+as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).
```
We'd better wait for confirmation of that publication date. Also, this is the wrong URL even if that date is right.

xmudrii replied:
Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

sftim reviewed (commented):
I just noticed that all the blog links are wrong (oops) and that we've not confirmed the date for the blog.

If we get to the date of release and this isn't resolved, we might opt to merge docs even with known errors. That would be OK if suboptimal.

/hold

xmudrii commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,13 +144,77 @@ For more information on version skews, see:
+  When upgrading to to a different minor release, you must bear in mind that
+  the package repository details also change.
+
+To learn more, you can read the official announcement ["pkgs.k8s.io: Introducing
+Kubernetes community-owned package repositories"](/2023/08/15/pkgs-k8s-io-introduction/).
```
@sftim I see, it's missing `/blog`. Is `/blog/2023/08/15/pkgs-k8s-io-introduction/` a correct URL given that the release date is correct? @ramrodo Can you please confirm if we want to do it on 2023-08-15 or earlier?

xmudrii replied:
Blog post link changed to `/blog/2023/08/15/pkgs-k8s-io-introduction/`

katcosgrove commented:
Hey y'all! I'm swinging by for the v1.28 Docs team. Today is the docs deadline. Is there anything we can help you with to get this merged?

sftim commented:
@katcosgrove I've kind of offered to help shepherd this one over the line (using my maintainer access to update xmudrii's fork if need be).

This is blocked on having a blog article publication date assigned I think.

Rishit-dagli commented:
> This is blocked on having a blog article publication date assigned I think.

And if I understand right that is blocked on having a final release date, right?

Rishit-dagli commented on content/en/docs/setup/production-environment/tools/kubeadm/install-kubeadm.md:
```diff
@@ -144,13 +144,77 @@ For more information on version skews, see:
+  the package repository details also change.
+
+To learn more, you can read the official announcement ["pkgs.k8s.io: Introducing
+Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).
```
As it seems the blog has been scheduled for 22 August
```suggestion
Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

sftim replied:
I think it makes sense to publish that on the 15th, and I'll see if release comms agree.

Rishit-dagli commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+This document assumes that you're already using the Kubernetes community-owned
+package repositories. If that's not the case, it's strongly recommend to migrate
+to the Kubernetes package repositories as described in
+["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).
```
```suggestion
["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

Rishit-dagli commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+
+**You're using the Kubernetes package repositories and this guide applies to you.**
+Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
+as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).
```
```suggestion
as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

Rishit-dagli commented on content/en/docs/tasks/administer-cluster/kubeadm/change-package-repository.md:
```diff
@@ -0,0 +1,164 @@
+
+**You're using the Kubernetes package repositories and this guide applies to you.**
+Otherwise, it's strongly recommend to migrate to the Kubernetes package repositories
+as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/15/pkgs-k8s-io-introduction/).
```
```suggestion
as described in ["pkgs.k8s.io: Introducing Kubernetes community-owned package repositories"](/blog/2023/08/22/pkgs-k8s-io-introduction/).
```

Rishit-dagli reviewed (changes requested):
Also requires corresponding changes in #42023

jeremyrickard commented:
/lgtm

sftim commented:
/hold cancel

sftim commented:
/lgtm
/approve

I'll send in a PR to _revert_ the last commit here.

sftim commented:
This tested fine previously, but Netlify hasn't updated the PR status to match.

sftim commented:
Helps with #42810

torenware commented:
PLEASE CONSIDER ROLLING THIS BACK.  Seriously.  You're screwing up the ability of users to install anything other than for v1.28, WHICH NEEDS TO BE DOCUMENTED IN THE CURRENT PAGE.

This is a screw-up, very simply.

cpanato commented:
> PLEASE CONSIDER ROLLING THIS BACK.  Seriously.  You're screwing up the ability of users to install anything other than for v1.28, WHICH NEEDS TO BE DOCUMENTED IN THE CURRENT PAGE.
> 
> 
> 
> This is a screw-up, very simply.

Please be more kind and respectful. If that breaks something on your end, let us know in a better way not using this type of language; most people here do this in their free time.
